release_dashboard/lib/widgets/merge_pr_substeps.dart

```diff
@@ -0,0 +1,221 @@
+class MergePrSubsteps extends StatefulWidget {
+  const MergePrSubsteps({
+    Key? key,
+    required this.nextStep,
+    required this.repository,
+  }) : super(key: key);
+
+  final VoidCallback nextStep;
+  final Repositories repository;
+
+  @override
+  State<MergePrSubsteps> createState() => MergePrSubstepsState();
+
+  static const Map<MergePrSubstep, String> substepTitles = <MergePrSubstep, String>{
+    MergePrSubstep.openPr: 'Open a pull request',
+    MergePrSubstep.updateLicenseHash: 'Update the license hash number',
+    MergePrSubstep.preSubmitCi: 'Validate pre-submit CI, have the PR reviewed, approved and merged',
+    MergePrSubstep.postSubmitCi: 'Validate post-submit CI',
+    MergePrSubstep.codesign: 'Codesign the engine binaries',
+  };
+
+  static const Map<MergePrSubstep, String> substepSubtitles = <MergePrSubstep, String>{
+    MergePrSubstep.openPr: 'You must now open a pull request by clicking on the following link: ',
+    MergePrSubstep.updateLicenseHash: 'There might be a license failure under the Linux Unopt test for this PR. '
+        "Visit the test's stdout, and there should be instructions on how to update the license hash number"
+        ' in the local checkout directory. After the update, please push the changes to the feature branch on your mirror.\n\n'
+        'This box can be checked if Linux Unopt passes the pre-submit CI.',
+    MergePrSubstep.preSubmitCi:
+        'Make sure all pre-submit tests on Github pass for this PR. Fix any test failures first. \n'
+            'Get the PR reviewed, approved and merged.',
+    MergePrSubstep.postSubmitCi: 'Make sure all post-submit CI tests pass here: ',
+    MergePrSubstep.codesign: 'An authorized person has to codesign the engine binaries.',
+  };
+
+  static const String releaseChannelInvalidErr = 'Error: Release Channel is invalid';
+
+  static const String noPrMsg = 'Since there are no code changes in this release, no PR is necessary.';
+}
+
+class MergePrSubstepsState extends State<MergePrSubsteps> {
+  final Map<MergePrSubstep, bool> _isEachSubstepChecked = <MergePrSubstep, bool>{};
+
+  @override
+  void initState() {
+    // If [substep] is false, that [substep] is unchecked, otherwise, it is checked.
+    // All substeps are unchecked at the beginning.
+    for (final MergePrSubstep substep in MergePrSubstep.values) {
+      _isEachSubstepChecked[substep] = false;
+    }
+    super.initState();
+  }
+
+  /// Toggle the boolean value of [substepName].
+  void substepPressed(MergePrSubstep substep) {
+    setState(() {
+      _isEachSubstepChecked[substep] = !_isEachSubstepChecked[substep]!;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final Map<ConductorStatusEntry, Object>? releaseStatus = context.watch<StatusState>().releaseStatus;
+    final ConductorService conductor = context.watch<StatusState>().conductor;
+    late bool isPrRequired;
+    late String newPrLink;
+
+    if (releaseStatus == null) {
```

Comment:
It is the same concept with [this](https://github.com/flutter/cocoon/pull/1487#discussion_r762354554)

Reply:
can you leave a TODO? this is something that would need to be cleaned up before this app can be released.

Reply:
Oh right, it is handled here: https://github.com/flutter/cocoon/pull/1500

Reply:
Done